### ISSUE TITLE
M1 Mac

### ISSUE THREAD

**Issue by bodohinz** [etesync/etesync-dav#229]
Can I use it on a M1 Mac?

**Comment by tasn** [comment]
Not with the pre-built binaries (not yet anyway).

**Comment by tasn** [comment]
Yeah, it would be great. I just need help with adding the github action. I dunno if that's supported for ARM yet on github actions.

**Comment by thomasfchang** [comment]
Just started using Etesync and I have an M1 Mac. It seems to work good at first sight through Rosetta 2.

An M1 Mac prompts you to install Rosetta the first time you want to run something that is built for x86. When you install Rosetta you can't uninstall it anymore unless you do a complete factory reset of your Mac. For users who would otherwise not have to use Rosetta (which is more and more the case since a majority of Mac software has been ported for M1 at this point) a native build would be really nice as not installing Rosetta would reduce your attack surface a lot by not being able to run x86 code.

**Comment by thomasfchang** [comment]
Unfortunately it does not seem to work anymore after a reboot

**Comment by Zvezdin** [comment]
Can't confirm - have it working fine on my end with Rosetta 2, just a resource hog.

**Comment by thomasfchang** [comment]
I'm trying to reinstall it now. Unfortunately now when I open the app and try to go to the web UI through http://localhost:37358/ it gives me an error that the page can't load

**Comment by tasn** [comment]
How bad of a resource hog?

As for a native build: defo, we just need to have CI that supports build on the M1. I think we use github actions and M1 isn't supported yet?

**Comment by Zvezdin** [comment]
It seems like we'll be [waiting for a while for that](https://github.com/actions/virtual-environments/issues/2187). Are there any alternative CIs you could use?

**Comment by thomasfchang** [comment]
Would it also not be possible to just release it as a regular Mac app with code signing? That way MacOS would not block the app on launch and then require you to explicitly grand permission to open the app in the security settings. That is what threw me off the first time trying Etesync. It might also make it possible for the app to make use of the MacOS [sandbox](https://developer.apple.com/library/archive/documentation/Security/Conceptual/AppSandboxDesignGuide/AboutAppSandbox/AboutAppSandbox.html) for better security, and building the app for M1 might be easier too

**Comment by thomasfchang** [comment]
I found out that the iPad app is available from the App Store to download on M1 Macs. I tried it out and the setup works flawlessly by giving the app access to Calendar, Reminders and Contacts. All the data is downloaded to the clients very fast without any problems. The only problem is that the app crashes when trying to sync manually after the setup. If this is a small problem that can be overcome easily then this would be a very user friendly option to run Etesync on arm Macs in my opinion. If the developer is interested I can give a crash report.

**Comment by pseacrest** [comment]
How did you get it to work, I keep getting 

Calendar: failed to find selected source 5DE3C489-C881-XXXXXX. Please contact developers.

> I found out that the iPad app is available from the App Store to download on M1 Macs. I tried it and the setup works flawlessly by giving the app access to Calendar, Reminders and Contacts. All the data is downloaded to the clients very fast without any problems. The only problem is that the app crashes when trying to sync manually after the setup. If this is a small problem that can be overcome easily then this would be a very user friendly option to run Etesync on arm Macs in my opinion. If the developer is interested I can give a crash report.

**Comment by pseacrest** [comment]
> How did you get it to work, I keep getting 
> 
> 
> 
> Calendar: failed to find selected source 5DE3C489-C881-XXXXXX. Please contact developers.
> 
> 
> 
> > I found out that the iPad app is available from the App Store to download on M1 Macs. I tried it and the setup works flawlessly by giving the app access to Calendar, Reminders and Contacts. All the data is downloaded to the clients very fast without any problems. The only problem is that the app crashes when trying to sync manually after the setup. If this is a small problem that can be overcome easily then this would be a very user friendly option to run Etesync on arm Macs in my opinion. If the developer is interested I can give a crash report.
> 
> 
> 
> 

Solved it myself by deleting said calendar and adding it a new

**Comment by benjaminrwilson** [comment]
@tasn, what changes would you recommend to build `etesync-dav` against `osx-arm64`?

**Comment by Zvezdin** [comment]
Hey @tasn, would github actions happen to support arm now?

**Comment by foliovision** [comment]
I'd love to see an M1 build for Apple Silicon Macs, as most of the time eteSync is the only Intel/Rosetta 2 application I have running. Just having Rosetta 2 open ties up memory and slows down an M1 Mac.

**Comment by tasn** [comment]
@Zvezdin, I don't think they do, no. :(

@foliovision, @benjaminrwilson: there are no changes needed on our end at all, it's just a matter of adding a CI task.
Additionally, etesync-dav is just Python. If you don't care about having a pre-built binary, and are OK with running Python code you can already run it on an M1 mac without Rosetta or anything.

**Comment by thomasfchang** [comment]
I have been using the iOS app on my Mac for a day now again and it seems to work fine. I don't know what caused the error when I first tried it. I actually like it more than the bridge because the bridge was quite buggy for me (sometimes not syncing things properly) and I like to be able to access my data within the app.

**Comment by foliovision** [comment]
> If you don't care about having a pre-built binary, and are OK with running Python code you can already run it on an M1 mac without Rosetta or anything.

Thanks @tasn very much for the suggestion. I can of course set up Python and set up a cron for eteSync. But as I'm paying (or will be paying) $24/year for privilege of access to eteSync, I don't want to have to think a lot about the internal workings of eteSync and maintaining it myself.

I.e. those of us on M1 Macs (and there will only be more of us, as many existing eteSync users migrate over to Apple Silicon: normally I'm not excited about new computers, before the M1 Max I was running 2009 Mac Pros and 2011 MBP 17" but these Apple Silicon MBP are finally silent, the Apple Studio is not, it has a constant fan running like an iMac) would like to have a pre-built binary which just works.

**Comment by tasn** [comment]
I understand @foliovision, I'd like to have it too. It's Github not having ARM build runners, not us being lazy.

**Comment by foliovision** [comment]
@tasn Tom, there has to be another way to build and attach M1 binaries. Most of the other open source or shareware projects that I either sponsor or license have M1 binaries. The great issue with programmers is that you assume that it's fair to outsource your work onto the user, instead of just providing working binaries which would save thousands of people from having to build such binaries themselves.

Do you see some money from a subscription to eteSync hosting or are you a separate project/entity?

**Comment by AlecKinnear** [comment]
This is Foliovision back, except on my personal account, as Alec Kinnear.

Since there's no ARM binary and eteSync runs constantly in the background (which means Rosetta2 has to run continuously wasting precious memory), I'm giving up testing eteSync and won't be subscribing. I'll have to go and make sure I haven't been billed now.

A real pity. eteSync is a great project, if you would support your users platforms with native versions. Let me know if you ever change this @tasn.

**Comment by tasn** [comment]
Thanks for the feedback @AlecKinnear, hopefully we get it sorted at some point.

**Comment by tasn** [comment]
> @tasn Tom, there has to be another way to build and attach M1 binaries. Most of the other open source or shareware projects that I either sponsor or license have M1 binaries. A difficult issue with programmers is that you assume that it's fair to outsource your work onto the user, instead of just providing working binaries which would save thousands of people from having to build such binaries themselves. Make users lives easier.

I've been toiling on this open source project away for years to make privacy and security better for everyone. Blaming me for not immediately jumping to support a platform I don't even have access to, when there are alternatives (Rosetta 2) available is just unfair. In a perfect world I'd have everything implemented for everyone immediately as they want it, but I can't do everything. I've been focusing on improvements to the server and infrastructure recently, and that's been taking my time.

It's open source though, everyone can change, improve, remix, and rebuild as they see fit. I'm not trying to push work to the user by saying this, I'm literally just pointing out the benefits of having all of the code open source.
The reason why you think programmers "assume it's fair to outsource your work onto the user", is because when you interact with programmers directly (rather than marketing/sales/whatever) it's with open source projects that actually ENABLE YOU to do all of that. If open source projects had sales people, you'd have heard them saying the same.

In short: I'm not trying to get users to do extra work, and again, if you've solved it and you think you have a solution that could benefit everyone, please share it. If you haven't, give us the benefit of the doubt and assume it's just a bit more work than you think it is.

**Comment by AlecKinnear** [comment]
@tasn I'm not asking for customisation or anything extraordinary. Apple M1 computers represent a large and growing market share. In particular people who owns these computers are those who will pay for software.

I'm not sure why it's so difficult for you to create binaries for M1 computers. Running Rosetta 2 in the background slows downs the computer and ties up a large amount of memory, just like running any VM within an OS.

As I said before, I'd like to sign up for the paid service of etedav but won't do it unless there's a binary for M1. You'd do better to just ask for a donation or something. I regularly donate to open source projects. We participate in open source projects which are primarily PHP or javascript based as that's where our expertise is. Rolling my own for etedav would be a real distraction.

@ tasn This attitude of pushing the work back at the user doesn't really work for hybrid open-source/commercial projects. You do have a paid service so it should be in your interest to add more paid users.

***

For those considering using etedav on an M1 Mac, don't do it until there's a proper binary. The etedav binary runs 24/7 so it's not something you want to have on Rosetta as it means you can never bring your M1 down to native apps only and Rosetta will always be running even when you are video editing, composing audio or doing large photo edits, slowing down your M1 and giving little IO hiccups when your creative app is running hard.

**Comment by AlecKinnear** [comment]
For those still struggling with an M1 Mac native solution, I think I found one which works, albeit it requires third-party software as the contacts client on macOS.

NextCloud will work with:

1. iOS devices Calendar and Contacts
2. BusyContacts ($30 with cross-grade, easy enough)
3. Thunderbird CardDAV (v.90+) (free)

BusyContacts is great for quickly cleaning up contacts.

NextCloud CardDAV will not work with macOS Contacts. NextCloud CalDAV does work with macOS Calendar.

Advantage over eteSync:  it's possible to share address books with other users in NextCloud.

Disadvantage: field labels are mangled by VCF export/import. eteSync doesn't mangle them. **eteSync is also more reliable**. When multiple changes collide on NextCloud with changes coming from two sources at the same time, NextCloud just prefers one whole set of changes instead of merging the changes. When the changes happen one at a time with no overlap in checkin/checkout, NextCloud seems to do just fine.

**Comment by Brog33** [comment]
Any news regarding an apple silicon build?

**Comment by AlecKinnear** [comment]
@Brog33 Sadly, the [release binary](https://github.com/etesync/etesync-dav/releases) has not been touched since December 2021.

Just circling around myself to see if I can/should subscribe to ete-dav. How many years of subscription has @tasn already given up by neglecting to create an M1 binary for ete-dav? It's a real pity. Open Source software is amazing and so are the people who make it. Unfortunately their stubbornness in not supporting users' platforms or enabling simple core features (CSV export for instance, insisting on .vcf as it's modern, hello @fruux) is their fatal weakness. This leaves commercial software more or less the entire field. In the case of ete-sync, it leaves spyware (Google, Apple, Microsoft) in control of the field.

For our own open-source projects, I offer those who request features not on the current immediate roadmap the opportunity to sponsor the new feature (partly as a test to see if people really want the feature instead and aren't just making noise).

**Comment by Zvezdin** [comment]
I'd pledge $10 if it helps getting this prioritised :). I'm a longtime user of @tasn's work and I appreciate the huge effort he's spent on impacting privacy. I understand that open source is a collaborative effort (with time or resources), so I am happy to put up a tip if it moves the needle. Really don't want to move to a less specialised solution such as Nextcloud, but etesync's the only thing running on rosetta on my mac.

**Comment by Michael-Gallo** [comment]
It seems Github has [ARM runners now](https://github.blog/news-insights/product-news/arm64-on-github-actions-powering-faster-more-efficient-build-systems/)

**Comment by AlecKinnear** [comment]
Since I'm a nice guy and I like to support open source, I have been paying for etesync for two years in the hope that @tasn Tom catches up with the real world and provides M1 binaries.

Open source creators unfortunately continue to be a disappointment. Even though Tom is taking our money, he can't be bothered to support what is one of the most popular computers in the world with designers and programmers.

@tasn Please take my money (you have it already) and ask someone to build an M1 binary. No need to wheedle further excuses about having to support the latest builds. My computers are running Monterey (I have other work to do than beta test Apple's poorly made yearly OS releases). You can lag behind by a full version (ideally etesync would have Sonoma support at this point). Those asking for Sequoia support would just have to wait.

**Comment by AlecKinnear** [comment]
Still paying (but not using etesync) and still getting the finger from Tom. Why do people dislike Open Source?

Because Open Source creators do not care about their users, only themselves and their own use case. Even if one pays most Open Source creators, they can't be bothered to help.

Thanks alot @tasn 

Don't cry me a river about how unappreciated your efforts are. I've been actively supporting your project for two years and you still can't be bothered to create a few lines of code for Apple Silicon.

**Comment by AlecKinnear** [comment]
I'm triply annoyed as there are issues with NextCloud right now, which I only use for Contact syncing (had been using file sharing but no longer) and would like to migrate to etesync. I thought that after two years, particularly with my monthly payments in his pocket (for no service), @tasn might get around to creating a version for Apple Silicon which does not require constantly running Rosetta 2 in the background.

Nope.

**Comment by tasn** [comment]
@AlecKinnear, I'm sorry to hear about your frustration, but I don't think the above statements are fair. Happy to share more context.
I don't think it's a one-line change (but if I'm wrong, please correct me and open a pull request with that line!), as it requires that all of the dependencies work well in ARM on Mac too, and python being python, that's not a given. So we are looking at quite a bit of time of doing the initial build. Add to that verification which will need to happen with someone (I don't even have an ARM mac), and also surely there are going to be issues that are going to surface so add more time fixing those.
The thing is, etesync-dav actually works fine on the Mac, it's just a question of (hopefully?) optimizing the memory usage, right? So by nature of that it's not a top priority, and there are other more important things, like the server code upgrades from a few months ago, the etesync.com servers from last month, etc.

Though as I said, if it's really that easy like you say, I'd love to get a patch so that we can close this ticket!

**Comment by AlecKinnear** [comment]
Hi Tom,

Thanks for your note.

> The thing is, etesync-dav actually works fine on the Mac, it's just a question of (hopefully?) optimizing the memory usage, right? So by nature of that it's not a top priority, and there are other more important things, like the server code upgrades from a few months ago, the etesync.com servers from last month, etc.

So etesync demands 2GB of memory to run on an M1 Mac. Good job, Tom.

> Add to that verification which will need to happen with someone (I don't even have an ARM mac)

At this point, ARM Macs don't cost very much (you can buy one for a few hundred dollars used, but I'd recommend the Mini 4 at $600 as with 16GB memory it's a very viable computer, not just a test platform, I'm writing this from the 32GB version which is now my main work computer, running off a large external boot disk).

I'm sure you know someone with an ARM Mac at this point (when I initially made the request, Apple Silicon was rarer and more expensive).

Everything is an obstacle for the man who doesn't want to go anywhere.

I'm contributing $3/month to etesync as a gesture of good faith with no use of it. Now you want me to develop it myself?

Thanks but we contribute to enough Open Source projects, either with donations or with code that we cannot take on another one, not related to our core expertise (PHP, online video). etesync is your project for which you collect money. You'd collect a heck of a lot more if it ran properly on modern Macs. Mac owners (not the recent ones, the historic ones) are the computer users most likely to pay for software.

Of course if your goal is to run etesync into the ground with a minimum viable effort, I understand.

Alec Kinnear

**Comment by Zvezdin** [comment]
@AlecKinnear I understand your concerns if you were talking about a SaaS business, but we're talking about @tasn's donated labour to the community here. If he were to be employed as a software engineer in NY where he's based, he'd make $100/hr minimum. The subscription moneys he receives is hardly going to compensate for the labour to maintain it, let alone add new stuff or to pay for all the work he's put to get to this place to begin with. 

So etesync is not a business - it is a gift to the community with a tiny revenue stream, and the community can make it even better if they'd like to chip in. Of course, you don't have to, and you're free to use other projects if you prefer. 

Just my 2c.

**Comment by AlecKinnear** [comment]
@Zvezdin 

> I understand your concerns if you were talking about a SaaS business, but we're talking about @tasn's donated labour to the community here. If he were to be employed as a software engineer in NY where he's based, he'd make $100/hr minimum.

Isn't that always the excuse for FOSS to be half-broken most of the time?

Why doesn't Tom @tasn take matters in hand and charge us all $5/month and improve etesync? etesync is a much needed utility (heaven knows Apple has let us all down with its horrible support for WebDAV, broken every other OS version).

Oh, yes, we love FOSS, we love broken software, we are donating our time. No, Tom's just taking up space with half-finished work.

> Of course, you don't have to, and you're free to use other projects if you prefer.

I donate plenty of time and money to open source myself. And I've donated to this project. I'd like sleepy Tom to wake up. He could hire someone to make these builds for less time than it takes to argue about them.

If he's short of cash, Tom should be asking me for bounty money and not whining that he doesn't feel like it. Since he's running a service, run it properly.

The suggestion to fix it myself was reasonable. I'd be happy to fix it myself, but our expertise is in design, PHP, old school JS, CSS, MySQL/MariaDB and not Python.

Until the FOSS community smartens up, the whole world must live with invasive, over-featured and expensive spyware. All of should get a clue and run our FOSS projects like a small business. Selling loaves of bread at a loss and complaining that you have no flour to bake more is absurd.

**Comment by tasn** [comment]
Thanks @Zvezdin for your kind note, I really appreciate it's exactly how I think about it.

@AlecKinnear, I've been an OSS maintainer for a very long time, so I have a thick skin. I'm slightly annoyed by your attitude, but it doesn't really bother me. However, at this point it's crossing the line to creating a toxic environment here in the community, so I'd please ask you to tone it down.

As for the topic at end: again, I'm sorry you're frustrated, and I wish it was easy to do what you want. In fact, someone tried in #340 a couple of days ago, but it's not as easy as you think. Though as I said, I'm happy to be proven wrong there.

**Comment by ponychicken** [comment]
What is the idea there?
